docs/contributing/contribute-testing.md

```diff
@@ -0,0 +1,70 @@
+# Testing guidelines
+
+k0smotron uses GitHub actions to run automated tests on all pull requests prior to merging.
+Pull request will not be reviewed before all tests are green,
+so to save time and prevent your Pull request from going stale,
+it is best to test it before submitting the pull request.
+
+## Run local verifications
+
+Run the following style and formatting commands to fix or check-in the changes:
+
+1. Verify code style using [`golangci-lint`](https://golangci-lint.run/). In the root directory of the repository run:
+
+   ```shell
+   make lint
+   ```
+
+   The build system automatically installs `golangci-lint`.
+
+2. Format Go source code using the `go fmt` command:
+
+
+   ```shell
+   go fmt ./...
+   ```
+
+3. Check documentation.
+
+   The Dockerized setup is used to locally perform documentation tests.
+   Run `make docs-serve-dev` to build documentation on http://localhost:8000.
+
+   If your port `8000` is busy, run `make docs-serve-dev DOCS_DEV_PORT=9999`.
+   The documentation page will be available on http://localhost:9999.
+
+4. Pre-submit flight checks.
+
+   In the repository root directory, ensure that the following commands are successfully run:
+
+    * `make build && git diff --exit-code`
+      This command verifies that the build is working
+      and that the generated source code matches the code checked into source control.
+    * `make test`
+      This command verifies that all the unit tests pass.
+
+   Once all tests have passed, you can open a pull request upstream.
+
+## Open pull request
+
+### Draft mode
+
+If you open a pull request in draft mode, it will only go through the automated testing.
+Once you decide the PR is ready for review, transition it out of draft mode to notify the k0smotron team.
+
+### Pre-requisites for merge
+
+In order for a pull request to be merged, the following conditions should exist:
+
+1. Pull request passes all the automated tests (style, build and conformance tests).
+2. Pull request commits are signed with the `--signoff` option.
+3. Pull request is reviewed and approved by a code owner.
+4. Pull request is rebased against upstream main branch.
```

Comment:
I suggest rethinking this edit.

First, can a PR be merged if any of these conditions are not fulfilled? The 'should' raises questions where perhaps questions should not be raised.

Second, if it is not a 'should' but is indeed a 'must', then the conditions should be presented in the past tense, as they were prior to this edit.

If the conditions are a 'must', I suggest:

For a pull request to be merged, all of the following must be true:

1. The PR has passed all of the automated tests (style, build and conformance tests).
2. All of the PR commits have been signed with the `--signoff` option.
3. The PR has been reviewed and approved by a code owner.
4. The PR has been rebased against upstream main branch.